### PR TITLE
Bluespace radio station fixed + improvements

### PR DESCRIPTION
**Pull request by carlopault**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl:
add: record disk burner can now be used to rename record disks
add: record disk burner now displays the name of the disk at the top of its main menu
del: Removed useless "linked_RS" variable
tweak: playing record disks on your PDA is less finnicky
fix: bluespace radio station now plays music like it's supposed to
/:cl:

image

## About The Pull Request
I have gotten around to fixing the radio station so that people can finally get the full DJ experience

## Why It's Good For The Game
DJ is cool.

### CODE REVIEW

**Comment by Zyzarda** [comment]
improvements GOOD +1

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -416,28 +415,39 @@
...
+					else
+						radio_holder = src.loc.loc.loc.loc.loc
+				else
+					radio_holder = src.loc.loc.loc.loc
+			else
+				radio_holder = src.loc.loc.loc
+		else
+			radio_holder = src.loc.loc
+	else
+		radio_holder = src.loc
```
Comment (yoyobatty) [inline code/game/objects/items/devices/radio/radio.dm]: Bro there's way better ways to check this, you could have atleast used a loop, this is excessive. Also src. can be removed, it's redundant

File: hippiestation/code/modules/library/lib_machines.dm [hunk 2]
```diff
@@ -213,7 +225,7 @@
 			return
 
 		qdel(I)
-		visible_message("<span class ='notice'>[user] loads a Spacestar-brand ink cartridge into the [src]")
+		src.visible_message("<span class ='notice'>[user] loads a Spacestar-brand ink cartridge into the [src]")
```
Comment (yoyobatty) [inline hippiestation/code/modules/library/lib_machines.dm:228]: Again, src. is not redundant and does nothing, I'm not sure why you added it to everything

**Review by yoyobatty — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (carlopault) [reply code/game/objects/items/devices/radio/radio.dm]: > Bro there's way better ways to check this, you could have atleast used a loop, this is excessive. Also src. can be removed, it's redundant

I tried making a loop but I wasnt creative enough so I brute forced it instead

File: hippiestation/code/modules/library/lib_machines.dm (hunk 2 above)
Comment (carlopault) [reply hippiestation/code/modules/library/lib_machines.dm:228]: > Again, src. is not redundant and does nothing, I'm not sure why you added it to everything

I saw src. on something somewhere else so I thought it was the right way to do it

Comment (yoyobatty) [reply hippiestation/code/modules/library/lib_machines.dm:228]: So you just looked and went oh wow I'm not going to question it, better add it? Sir, that's hasty and shoves a big fuck you to basic troubleshooting. Clearly nothing was broken so there's no point to this.

Comment (carlopault) [reply hippiestation/code/modules/library/lib_machines.dm:228]: > So you just looked and went oh wow I'm not going to question it, better add it? Sir, that's hasty and shoves a big fuck you to basic troubleshooting. Clearly nothing was broken so there's no point to this.

yeah

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (carlopault) [reply code/game/objects/items/devices/radio/radio.dm]: couldnt think of any way to make a loop work

**Comment by carlopault** [comment]
too easy

**Review by yoyobatty — APPROVED** [review]

**Pull request merged**